### PR TITLE
Add search (filter) for an specific Integration on Homepage's Integrations section 

### LINKED ISSUE #1798: Add search (filter) for an specific Integration on Homepage's Integrations section
### Current Behavior

On https://meshery.io/ homepage  a section displays all the integrations that the project supports.
### Desired Behavior

[meshery.io](http://meshery.io/)’s home page is in need of a search (filter) for site visitors who wish to search for a specific integration.<br>

### Screenshots / Mockups

Screenshot 2024-07-03 at 2 19 02 PM
---
<h4>Contributor [Guides](https://docs.meshery.io/project/contributing) and Resources</h4>

- 🎨 Wireframes and [designs for Meshery UI](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI) in Figma [(open invite)](https://www.figma.com/team_invite/redeem/qJy1c95qirjgWQODApilR9)
- 🖥 [Contributing to Meshery Website](https://github.com/meshery/meshery.io#contributing-to-the-mesheryio-website)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](https://discuss.layer5.io) and [Community Slack](https://slack.meshery.io)

### PR DESCRIPTION
**Pull request by shubhusion**
**Description**

This PR fixes #1798

**Notes for Reviewers**

**[Signed commits](https://docs.meshery.io/project/contributing#signing-off-on-commits-developer-certificate-of-origin)**
- [X] Yes, I signed my commits.

### CODE REVIEW

**Comment by shubhusion** [comment]
@vishalvivekm can you please review this PR ?

**Comment by Ashparshp** [comment]
@shubhusion Thanks for your contribution, let's discuss this on the website's call. Please add this as an agenda item to the [meeting minutes](https://docs.google.com/document/d/1XczAHXVe2FIWPqiF57ospJ43zw5cZQ7ui8mn39v5EvA/edit?usp=drivesdk). The call is scheduled for Monday at 5:30 PM IST.

**Comment by Ashparshp** [comment]
@shubhusion Thanks for the words!

**Comment by shubhusion** [comment]
@Ashparshp can you review this PR ? Earlier @hargunkaur286 had reviewed it.

**Comment by shubhusion** [comment]
@vishalvivekm required changes has been made.

**Review by hargunkaur286 — APPROVED** [review]

**Comment by amitamrutiya** [comment]
I found several issues while reviewing your PR. I'm not sure if these are only happening on my end. 

[Screencast from 2024-08-12 09-50-39.webm](https://github.com/user-attachments/assets/01bd215e-e4b0-42e0-be59-200e48bd8245)

I am not able to see the cursor icon on the button and can only click certain areas on the buttons below. Other issues are mentioned in the video. Also, the gap between buttons is different from our previous one.

I am using Firefox for testing.

**Comment by shubhusion** [comment]
> I found several issues while reviewing your PR. I'm not sure if these are only happening on my end. 
> 
> [Screencast from 2024-08-12 09-50-39.webm](https://github.com/user-attachments/assets/01bd215e-e4b0-42e0-be59-200e48bd8245)
> 
> I am not able to see the cursor icon on the button and can only click certain areas on the buttons below. Other issues are mentioned in the video. Also, the gap between buttons is different from our previous one.
> 
> I am using Firefox for testing.

@amitamrutiya i have made no modifications in buttons. My task was to add a search filter above the icons.

**Comment by amitamrutiya** [comment]
I think this PR makes the above mentioned changes.

**Comment by shubhusion** [comment]
> I think this PR makes the above mentioned changes.

@amitamrutiya buttons are working for me.
spacing and other issues i will rectify

https://github.com/user-attachments/assets/18dc38fd-2ac0-4ef6-adf3-b1ccb2327225

**Comment by shubhusion** [comment]
@amitamrutiya all issues you raised have been rectified.

**Comment by shubhusion** [comment]
@vishalvivekm @sudhanshutech please review this pr

**Review by Ashparshp — COMMENTED** [review]
We can fix those issues as well, as @amitamrutiya pointed out. However, let's address only the ones that have been modified for now. We can proceed with the PR if @shubhusion is not going to work on the fix. @amitamrutiya, you can take up the rest in a separate issue if you'd like.

@shubhusion thoughts?

**Comment by shubhusion** [comment]
> LGTM! We can fix those issues as well, as @amitamrutiya pointed out. However, let's address only the ones that have been modified for now. We can proceed with the PR if @shubhusion is not going to work on the fix. @amitamrutiya, you can take up the rest in a separate issue if you'd like.
> 
> @shubhusion thoughts?

@Ashparshp The issues pointed out by @amitamrutiya have been rectified in the last two commits.

**Comment by Ashparshp** [comment]
Thank you @shubhusion.

**Comment by shubhusion** [comment]
@hargunkaur286 can you please review this pr ?

File: _includes/partials/homePage-integrations.html [hunk 1]
```diff
@@ -1,83 +1,166 @@
+<style>
```
Comment (sudhanshutech) [inline _includes/partials/homePage-integrations.html:1]: @shubhusion why the existing search bar that is in catalog page is being not reused? Why we have to create new component and styles? Please reuse that

**Review by sudhanshutech — CHANGES_REQUESTED** [review]

File: _includes/partials/homePage-integrations.html (hunk 1 above)
Comment (sudhanshutech) [reply _includes/partials/homePage-integrations.html:1]: @shubhusion i didn't heard from you...

Comment (shubhusion) [reply _includes/partials/homePage-integrations.html:1]: I will make the necessary changes, but I'm a bit busy this week. Please give me some time.

Comment (shubhusion) [reply _includes/partials/homePage-integrations.html:1]: @shubhusion 

> @shubhusion why the existing search bar that is in catalog page is being not reused? Why we have to create new component and styles? Please reuse that

The styles from the existing search bar on the catalog page were duplicated and adapted for the new search bar on the home page. Some modifications were made to ensure it fits well on the home page.

To reuse the same styles, I will need to create a shared .css file that is accessible to both pages.

**Comment by Ashparshp** [comment]
@shubhusion, I'm adding this as an agenda item to the meeting minutes. Please join the evening call and let's take it forward.

**Pull request closed without merging**